presto-common/src/main/java/com/facebook/presto/common/RuntimeMetric.java

```diff
@@ -39,28 +41,46 @@
      * Creates a new empty RuntimeMetric.
      *
      * @param name Name of this metric. If used in the presto core code base, this should be a value defined in {@link RuntimeMetricName}. But connectors could use arbitrary names.
+     * @param unit Unit of this metric. Available units are defined in {@link RuntimeUnit}.
      */
-    public RuntimeMetric(String name)
+    public RuntimeMetric(String name, RuntimeUnit unit)
     {
         this.name = requireNonNull(name, "name is null");
+        this.unit = requireNonNull(unit, "unit is null");
+    }
+
+    public RuntimeMetric(String name)
+    {
+        this(name, NONE);
     }
 
     public static RuntimeMetric copyOf(RuntimeMetric metric)
     {
         requireNonNull(metric, "metric is null");
-        return new RuntimeMetric(metric.getName(), metric.getSum(), metric.getCount(), metric.getMax(), metric.getMin());
+        return new RuntimeMetric(metric.getName(), metric.getUnit(), metric.getSum(), metric.getCount(), metric.getMax(), metric.getMin());
+    }
+
+    public RuntimeMetric(
+            String name,
+            long sum,
+            long count,
+            long max,
+            long min)
+    {
+        this(name, NONE, sum, count, max, min);
     }
 
     @JsonCreator
     @ThriftConstructor
     public RuntimeMetric(
             @JsonProperty("name") String name,
+            @JsonProperty("unit") RuntimeUnit unit,
             @JsonProperty("sum") long sum,
             @JsonProperty("count") long count,
             @JsonProperty("max") long max,
             @JsonProperty("min") long min)
     {
-        this(name);
+        this(name, unit);
```

Comment:
Since the older version would pass in null as the `unit`, would it throw exceptions because of the `requireNonNull ` check in construction function?

Reply:
It will be checked inside `this(name, unit)`.

Reply:
Yes, and it throws exception, so wouldn't the new version of presto that has this change be incompatible with older version?